state/executor.go

```diff
@@ -348,10 +346,13 @@ func (t *Transition) GetTxnHash() types.Hash {
 }
 
 // Apply applies a new transaction
-func (t *Transition) Apply(msg *types.Transaction) (*runtime.ExecutionResult, error) {
+func (t *Transition) Apply(
+	msg *types.Transaction,
+	tracer runtime.Tracer,
```

Comment:
I am not comfortable changing the `Apply` function directly. I think `Tracer` should be an optional object for the `Transition` and the specific `Tracer` instance should figure out if it only wants to track a specific transaction.

Reply:
2a5da666eb66487793d7379d3f3683b591df486b